packages/grafana-ui/src/components/Combobox/Combobox.tsx

```diff
@@ -124,6 +125,10 @@ export const Combobox = <T extends string | number>(props: ComboboxProps<T>) =>
 
   // A custom setter to always prepend the custom value at the beginning, if needed
   const [items, baseSetItems] = useState(isAsync ? [] : options);
+  // To save the options as long as the group items to generate rows per each
+  const [virtualizedItems, setVirtualizedItems] = useState<Array<ComboboxOption<T> | ComboboxOption<T>['group']>>(
```

Comment:
Why can't we use baseSetItems for this?

Reply:
I thought we wanted to preserve the option list without being modified and just with options.